x-pack/plugins/security_solution/public/common/components/events_tab/events_query_tab_body.tsx

```diff
@@ -86,6 +99,52 @@ const EventsQueryTabBodyComponent: React.FC<EventsQueryTabBodyComponentProps> =
   const { globalFullScreen } = useGlobalFullScreen();
   const ACTION_BUTTON_COUNT = 5;
   const tGridEnabled = useIsExperimentalFeatureEnabled('tGridEnabled');
+  const [defaultNumberFormat] = useUiSetting$<string>(DEFAULT_NUMBER_FORMAT);
+  const [showExternalAlerts, setShowExternalAlerts] = useState<boolean>(false);
+  const leadingControlColumns = useMemo(() => getDefaultControlColumn(ACTION_BUTTON_COUNT), []);
+
+  const toggleExternalAlerts = useCallback(() => setShowExternalAlerts((s) => !s), []);
+
+  const getSubtitle = useCallback(
+    (totalCount: number) =>
+      `${SHOWING}: ${numeral(totalCount).format(defaultNumberFormat)} ${UNIT(totalCount)}`,
+    [defaultNumberFormat]
+  );
+
+  const histogramExtraProps = useMemo(
+    () => ({
+      ...(showExternalAlerts
+        ? {
+            ...alertsHistogramConfig,
+            id: ALERT_HISTOGRAM_ID,
+            defaultModel: alertsDefaultModel,
+            pageFilters: externalAlertPageFilters,
+            subtitle: getSubtitle,
+          }
+        : {
+            ...histogramConfigs,
+            id: EVENTS_HISTOGRAM_ID,
+            defaultModel: eventsDefaultModel,
+            pageFilters,
+            unit,
+          }),
+    }),
+    [externalAlertPageFilters, getSubtitle, pageFilters, showExternalAlerts]
+  );
+
+  const statefulEventsViewerExtraProps = useMemo(
+    () => ({
+      ...(showExternalAlerts
+        ? {
+            pageFilters: [defaultAlertsFilters, ...(externalAlertPageFilters || [])],
```

Comment:
spread missing? `...defaultAlertsFilters`

Reply:
nah, defaultAlertFilters is an object already. pageFilters is an array of Filter objects